apicast/bin/apicast

```diff
@@ -110,6 +110,7 @@ done
 
 main+=("daemon ${daemon};")
 main+=("worker_processes ${worker_processes};")
+main+=$(printenv | awk '$1 ~ /^(APICAST|THREESCALE)_/ {split($0,env,"="); print "env", env[1] ";"}')
```

Comment:
To be honest, I don't like this approach too much, for the following reasons:

1) It makes `bin/apicast` the only way to start APIcast that would work natively out-of-the-box. Even though it is the preferred way to start APIcast, it might not suite some users who have some already established deployment pipelines (Chef, Puppet etc, where OpenResty is started as a service). Well, yes, you can of course add all the required env definitions in `main.d`, but it's easy to forget something there.

2) This way I think we are loosing the "single source of truth". It's actually super practical to have all the supported env vars listed in one place. We do have [.env](https://github.com/3scale/apicast/blob/master/.env) and [parameters doc](https://github.com/3scale/apicast/blob/master/doc/parameters.md) that can help, but as you usually (rightfully) say @mikz - the docs can get out of sync (the `.env` already did, for example) :)
So, it would actually be good to keep the complete list in some file, that can be extended with other files.